tests/pthread/test_pthread_proxying_refcount.c

```diff
@@ -55,26 +57,35 @@ void* execute_and_free_queue(void* arg) {
     em_proxying_queue_destroy(queues[i]);
   }
 
+  // Wrap the normal worker event listener so that we can determine when our
+  // proxying events have been received and handled.
+  EM_ASM({
+      var oldOnMessage = onmessage;
+      onmessage = (e) => {
+        oldOnMessage(e);
+        if (e.data.cmd == 'processProxyingQueue') {
+          _register_processed();
+        }
+      };
+    });
+
   // Exit with a live runtime so the queued work notification is received and we
-  // try to execute the queue again, even though it has been destroyed.
+  // try to execute the queue again, even though we already executed all its
+  // work and we are now just waiting for the notifications to be received so we
+  // can free it.
   emscripten_exit_with_live_runtime();
 }
 
 int main() {
+  emscripten_console_log("start");
```

Comment:
```suggestion
```

Reply:
Adding this printing back in simplifies the overall diff, since they were originally removed as part of this PR. (My guess is that you only looked at the latest diff, where this looks like an unnecessary change?)

Reply:
Ah, yes. Lgtm either way, then.